review_app_status.py

```diff
@@ -0,0 +1,137 @@
+# -*- encoding: utf-8
+
+import json
+import logging
+import os
+import time
+from enum import Enum
+
+import requests
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger("review_app_status")
+
+
+class BuildStates(Enum):
+    """Review App Build states"""
+
+    success = "success"
+
+
+def _make_github_api_request(url):
+    """Perform Github API Queries.
+
+    Input:
+        url: URL to fetch.
+    Output:
+        JSON Response.
+    """
+    headers = {
+        "Accept": "application/vnd.github.v3+json",
+        "Authorization": f"token {os.environ['GITHUB_TOKEN']}",
+    }
+
+    r = requests.get(url, headers=headers)
+    if r.status_code != requests.codes.ok:
+        r.raise_for_status()
+    return r.json()
+
+
+def _get_github_deployment_status_url(deployments_url, commit_sha):
+    """Get deployment_status URL for the head commit.
+        Inputs:
+            deployments_url: This can be obtained from `pull_request` event payload.
+            commit_sha: SHA of head/latest commit. This  also can be obtained from `pull_request` event payload.
+        Output:
+            deployment status URL.
+    """
+    deployments = _make_github_api_request(deployments_url)
+    for deployment in deployments:
+        if deployment["sha"] == commit_sha:
+            return deployment["statuses_url"]
+
+    raise ValueError("No deployment found for the lastest commit.")
+
+
+def _get_build_data(url, interval):
+    """Get Review App build data using Github's `deployment_status` API.
+    
+        Inputs:
+            url: deployment status URL
+            interval: Amount of time (in seconds) to check the build status 
+                    if the status is in pending state.
+        
+        Output:
+            Review App build data
+    """
+    while True:  
+        response = _make_github_api_request(url)
+
+        # Heroku returns empty list until the build is succeeded/Failed.
+        if len(response) < 1: # pragma: no cover
```

Comment:
use `side_effect` to mock different values for subsequent calls then you can test this

Reply:
Done